src/phoenix/server/api/routers/v1/datasets.py

```diff
@@ -518,50 +457,87 @@ async def _parse_form_data(
+      200:
+        description: Success
+        content:
+          text/csv:
+            schema:
+              type: string
+              contentMediaType: text/csv
+              contentEncoding: gzip
+      404:
+        description: Dataset does not exist.
+      422:
+        description: Dataset ID or version ID is invalid.
+    """
+    if id_ := request.path_params.get("id"):
+        try:
+            dataset_id = from_global_id_with_expected_type(GlobalID.from_id(id_), Dataset.__name__)
+        except ValueError:
+            return Response(
+                content=f"Invalid Dataset ID: {id_}",
+                status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+    else:
         return Response(
             content="Missing Dataset ID",
             status_code=HTTP_422_UNPROCESSABLE_ENTITY,
         )
-    dataset_global_id = GlobalID.from_id(encoded_dataset_id)
-    if dataset_global_id.type_name != Dataset.__name__:
-        return Response(
-            content=f"Invalid Dataset ID: {encoded_dataset_id}",
-            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
-        )
-    encoded_dataset_version_id = request.path_params.get("datasetVersionId")
-    dataset_version_global_id = (
-        GlobalID.from_id(encoded_dataset_version_id) if encoded_dataset_version_id else None
-    )
-    if dataset_version_global_id and dataset_version_global_id.type_name != DatasetVersion.__name__:
-        return Response(
-            content=f"Invalid Dataset Version ID: {encoded_dataset_version_id}",
-            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
-        )
-    dataset_id = int(dataset_global_id.node_id)
+    dataset_version_id: Optional[int] = None
+    if version := request.query_params.get("version"):
+        try:
+            dataset_version_id = from_global_id_with_expected_type(
+                GlobalID.from_id(version),
+                DatasetVersion.__name__,
+            )
+        except ValueError:
+            return Response(
+                content=f"Invalid Dataset Version ID: {version}",
+                status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            )
     mder = models.DatasetExampleRevision
     latest_version = (
         select(
             mder.dataset_example_id,
-            (func.rank().over(order_by=mder.dataset_example_id) - 1).label("example_index"),
             func.max(mder.dataset_version_id).label("dataset_version_id"),
         )
         .group_by(mder.dataset_example_id)
         .join(models.DatasetExample)
         .where(models.DatasetExample.dataset_id == dataset_id)
     )
-    if dataset_version_global_id:
-        max_dataset_version_id = int(dataset_version_global_id.node_id)
+    if dataset_version_id is not None:
+        max_dataset_version_id = (
+            select(models.DatasetVersion.id)
+            .where(models.DatasetVersion.id == dataset_version_id)
+            .where(models.DatasetVersion.dataset_id == dataset_id)
+        ).scalar_subquery()
```

Comment:
If I am understanding correctly, the purpose of this subquery is to ensure that the dataset version actually exists, but if the dataset version does not exist, we'll be returning the misleading error message `Dataset does not exist.` We should clarify the language in the error message, e.g., `Could not identify dataset given id ... and version ...` or we should make follow-up queries like we do [here](https://github.com/Arize-ai/phoenix/blob/ff9a475b303a63d7f63b8f895a4716c0064e64d1/src/phoenix/server/api/routers/v1/dataset_examples.py#L74) to return more granular error messages.

Reply:
If the version does not exist, it returns an empty CSV.

Also, I didn't want to add a separate query just for an edge case.

Reply:
`Dataset does not exist.` is returned only when the dataset name can't be determined (it may be deleted)